routes/login.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Created by Trevor on 6/26/2015.
+ */
+
+var express = require("express");
+var router = express.Router();
+var Q = require('q');
+var crypto = require('crypto');
+var uuid = require('node-uuid');
+
+router.route('/').post( function(req,res){
+
+    var username=req.body.user;
+    var password=req.body.password;
+
+    var db = require("../imp_services/impdb.js").connect();
+    Q.fcall(db.beginTransaction())
+        .then(db.query("USE " + db.databaseName))
+        .then(db.query("CALL GetUserByUsername(\'" + username + "\');"))
+        .then(function(row) {
+
+           // We got data about the user
+            if (row[0][0].length == 0) { // No user by that username
+                res.end("Invalid Credentials!");
+            }
+            else {
+            var oldhash = row[0][0][0].HP;
+                var salt = row[0][0][0].US;
+                var hash = crypto.createHash('sha256').update(password + salt).digest('hex');
+
+                if (hash == oldhash) {
+                    var cookie = uuid.v4();
+                    console.log("Hash match!");
+                    res.cookie('auth', cookie, {secure: false, maxAge: 60 * 1000, httpOnly: false});
```

Comment:
please name the cookie "IMPId", the user can have potentially several cookies stored from different sites in their browser at any given time, we want to make ours specific to our app so as not to collide with other website cookies.